service/history/workflow/mutable_state_impl.go

```diff
@@ -6769,6 +6788,14 @@ func (ms *MutableStateImpl) updatePendingEventIDs(
 			continue
 		}
 	}
+	if len(requestIDToEventID) > 0 {
+		for requestID, eventID := range requestIDToEventID {
+			if requestIDInfo, ok := ms.executionState.RequestIds[requestID]; ok {
+				requestIDInfo.EventId = eventID
+			}
+		}
+		ms.executionStateUpdated = true
```

Comment:
Synced offline. Let's not set this flag when updating RequestIds (here and in `attachRequestID`). 
It is meant to only capture changes of the State/Status field in executionState, otherwise standby side will refresh workflow start related tasks unnecessarily. 

I will send out a follow up PR to rename those fields.

Reply:
Removed